### ISSUE TITLE
AppendBlock from azure-sdk-for-go v50.1.0 doesn't work

### ISSUE THREAD

**Issue by mihaitodor** [Azure/Azurite#682]
### Which service(blob, file, queue, table) does this issue concern?

Blob

### Which version of the Azurite was used?

Latest (v3.10.0)

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)

DockerHub (`mcr.microsoft.com/azure-storage/azurite:3.10.0`)

### What's the Node.js version?

Whatever the `mcr.microsoft.com/azure-storage/azurite:3.10.0` docker container is running

### What problem was encountered?

[`Blob. AppendBlock`](https://github.com/Azure/azure-sdk-for-go/blob/ac17097a799300c6446ec0e3f899a4f0b2a81375/storage/appendblob.go#L70) from `github.com/Azure/azure-sdk-for-go` `v3.10.0` fails with error: `Content-Length must be 0 for Create Append Blob request.`

### Steps to reproduce the issue?

Run the following code:

```go
package main

import (
	"log"
	"time"

	"github.com/Azure/azure-sdk-for-go/storage"
	"github.com/ory/dockertest/v3"
	"github.com/ory/dockertest/v3/docker"
)

// RunAzurite starts an azurite container
func RunAzurite(pool *dockertest.Pool) (*dockertest.Resource, error) {
	opts := dockertest.RunOptions{
		Repository:   "mcr.microsoft.com/azure-storage/azurite",
		Tag:          "3.10.0",
		ExposedPorts: []string{"10000"},
		PortBindings: map[docker.Port][]docker.PortBinding{
			"10000": {{HostIP: "0.0.0.0", HostPort: "10000"}},
		},
	}
	azurite, err := pool.RunWithOptions(&opts)
	if err != nil {
		return nil, err
	}
	if eerr := azurite.Expire(10); eerr != nil {
		return nil, eerr
	}
	pool.MaxWait = 10 * time.Second
	rerr := pool.Retry(func() error {
		client, eerr := storage.NewEmulatorClient()
		if eerr != nil {
			return eerr
		}
		s := client.GetBlobService()
		c := s.GetContainerReference("cont")
		if _, err = c.Exists(); err != nil {
			return err
		}
		return nil
	})
	return azurite, rerr
}

func main() {
	pool, err := dockertest.NewPool("")
	if err != nil {
		log.Fatalf("Failed to create dockertest pool: %s", err)
	}

	azurite, err := RunAzurite(pool)
	defer pool.Purge(azurite)

	if err != nil {
		log.Fatalf("Failed to start Azurite: %s", err)
	}

	client, err := storage.NewEmulatorClient()
	if err != nil {
		log.Fatalf("Failed to create storage client: %s", err)
	}

	blobClient := client.GetBlobService()
	dummyContainer := "foobar"
	containerRef := blobClient.GetContainerReference(dummyContainer)

	err = containerRef.Create(nil)
	if err != nil {
		log.Fatalf("Failed to create container: %s", err)
	}

	blobRef := containerRef.GetBlobReference("test.txt")

	err = blobRef.PutAppendBlob(nil)
	if err != nil {
		log.Fatalf("Failed to create append blob: %s", err)
	}

	dummyData := "deadbeef"
	err = blobRef.AppendBlock([]byte(dummyData), nil)
	if err != nil {
		log.Fatalf("Failed to write to append blob: %s", err)
	}

	log.Print("That's all folks!")
}
```

[This check](https://github.com/Azure/Azurite/blob/8c61bf42e8fb5d2d8fdff9f8e14f1bb9bf34ecde/src/blob/handlers/AppendBlobHandler.ts#L32-L37) seems to be the root cause.

### Have you found a mitigation/solution?

I guess it works with `github.com/azure/azure-storage-blob-go` but I can't use it because it doesn't yet come with support for connection strings (see Azure/azure-storage-blob-go#211 and Azure/azure-storage-blob-go#244).

**Comment by blueww** [comment]
@mihaitodor 
For the [This check ](https://github.com/Azure/Azurite/blob/8c61bf42e8fb5d2d8fdff9f8e14f1bb9bf34ecde/src/blob/handlers/AppendBlobHandler.ts#L32-L37) you shared, the check to for create append blob the content-length should be 0. This is correct, from [rest API doc](https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob), you can see `For a page blob or an append blob, the value of this header must be set to zero, as Put Blob is used only to initialize the blob.`

Not sure if your failure is caused by other issues.
Would you please share the Azurite debug log for this issue, then we will be more clear on this failure?

**Comment by mihaitodor** [comment]
@blueww Thank you for looking into this!

Indeed, the legacy `azure-sdk-for-go` does not adhere to the docs in this case and the Azure backend happily accepts this behaviour without enforcing the spec. See the code [here](https://github.com/Azure/azure-sdk-for-go/blob/d06c8bd2777ec3e2a6e4b0457abfb3052b6c95eb/storage/appendblob.go#L74) in `AppendBlock`, where `headers["Content-Length"] = fmt.Sprintf("%v", len(chunk))`.

Here is the Azurite debug log:
[debug.log](https://github.com/Azure/Azurite/files/5904517/debug.log)

I also tried passing in the `--loose` flag to azurite, but it didn't prevent this error (although maybe it should?).

This is a bug in `azure-sdk-for-go`, but I guess that won't change (I can mirror the bug report there if you wish), and apps that still make use of this SDK and call `AppendBlock` can't be tested using Azurite.

**Comment by blueww** [comment]
@mihaitodor
This looks an legacy SDK Bug, I have checked other SDK (.net), and it will set content-length as 0.
Since you use legacy  SDK, it's very possible the bug won't be fix on the legacy SDK.
Would you like to upgrade to the latest Go SDK, and if you still see the bug, can ask the SDK team to fix it?

Besides that, we welcome contribution!
If you don't want to upgrade SDK, as a workaround, you can raise a PR to skip this check in --loose mode of Azurite.

**Comment by mihaitodor** [comment]
@blueww Thanks for the feedback! Unfortunately, as mentioned before, it's not (yet) feasible to upgrade to the new Go SDK because it doesn't have support for connection strings...

I have opened #698 to skip this check when the `--loose` flag is set.

**Comment by blueww** [comment]
The fix is already released in v3.11.0.
